File: 3765-complete-prime-number/3765-complete-prime-number.java
class Solution {
    private HashMap<Integer, Boolean> p = new HashMap<>();
    private Boolean isPrime(int x) {
        if (p.containsKey(x)) return p.get(x);
        if (x == 0) return true;
        if (x == 1) return false;
        for (int i = 2; i <= Math.sqrt(x); i++) {
            if (x % i == 0) {
                p.put(x,false);
                return false;
            }
        }
        p.put(x,true);
        return true;
    }
    public boolean completePrime(int num) {
        if (!isPrime(num)) return false;
        int tn = 10;
        while (tn < num) {
            int x = num % tn;
            tn *= 10;
            if (!isPrime(x)) return false;
        }
        while (tn/10 > 0) {
            int x = num / tn;
            tn /= 10;
            if (!isPrime(x)) return false;
        }
        return true;
    }
}

File: 3765-complete-prime-number/3765-complete-prime-number.py
p = {}
class Solution:
    def isPrime(self, x):
        if x in p:
            return p[x]
        if x == 1:
            return False
        for i in range(2,int(x**(0.5)) + 1):
            if x % i == 0:
                p[x] = False
                return False
        p[x] = True
        return True
    def completePrime(self, num: int) -> bool:
        x = 0
        tn = 10
        while tn <= num:
            x = num % tn
            tn *= 10
            if not self.isPrime(x):
                return False
        x = 0
        tn = 10**len(str(num))
        while tn > 0:
            x = num // tn
            tn //= 10
            if not self.isPrime(x):
                return False
        return True